### ISSUE TITLE
Custom command to hide passwords under notes does not work in 8.0.0

### ISSUE THREAD

**Issue by setedivento** [hluk/CopyQ#2725]
**Context**
what I need to do is to make CopyQ, when the main window is displayed, hide the text of those clipboard items that have an associated note – provided they are placed under a specific tab.
Let's say the tab is called "auths" and I have all my stored passwords in there: what I want to achieve is being able to copy the passwords as regular text as usual, but to hide them when, suppose, I have to share the screen during work-calls.

**Bug description**
I have created a custom command to achieve what I want, and that has always worked in past versions of CopyQ: while now, whether you install the same version of CopyQ (8.0.0) on Windows 10 or Windows 11, it doesn't make a difference, and it works only in a particular case, but not the main desidered one. 
The command's script is conceived to work in Display mode, so it only affects how copy items are shown, and not their clipboard content when they are copy-pasted.

To look at the script's details, please refer to the following images.
![1 script code](https://github.com/hluk/CopyQ/assets/20049455/b1602c83-193b-4712-802a-ce37612920ca)
![2 script cfg](https://github.com/hluk/CopyQ/assets/20049455/742e5da6-cb13-4b18-883b-a15f30137a00)

**To Reproduce**
please add a new Command with this code:

```typescript
copyq:
var notes = str(data(mimeItemNotes))
if (notes) {
  setData(mimeText, "")
}
```
 
and configure the Advanced options like in the image above, creating a new tab and setting it as the target of the Command.
Now in the dedicated tab you have created, add a new copy item, and then add a note to that: let's call this item "item with a note".

**Expected behavior**
The item with a note, when the main window of the program in activated, should display the note text only, and not its original source text; but then when the item is pasted, CopyQ should paste the source text as usual.

**Screenshots**
this is the result of the Command when the main program window is dispayed and it works
![notes_working](https://github.com/hluk/CopyQ/assets/20049455/f1c448a2-6455-4a61-a494-a1bf53c18f2e)
this is the result of the Command when the main program window is dispayed and it doesn't work
![notes_not_working](https://github.com/hluk/CopyQ/assets/20049455/31126207-2519-4e32-94ec-51c11935afd8)

**Version, OS and Environment**
- Application Version: 8.0.0
- OS: Windows 10, Windows 11
- Desktop environment, normal explorer.exe shell

**Additional notes**
the Command works when you modify the notes of an item; but the change is not persistent, and all the items return to their previous state when one exits and opens the program again.

**Thanks so much for fixing this!**

**Comment by hluk** [comment]
I'll look into this.

Anyway, the best way to hide the item content is to use `setData(mimeHidden, '1')` instead.

**Comment by hluk** [comment]
I cannot reproduce the behavior with `setData`. I think you need to hide also other text types:

```js
setData(mimeText);
setData(mimeHtml);
setData(mimeTextUtf8);
```

It is better to use `setData(mimeHidden, '1')`.

**Comment by setedivento** [comment]
Ow, thanks for the explanation.
Anyway it was solved with 9.0.0, I thought when in the release note you wrote:
_ @Fixes situation when display commands stop updating items._
you meant this bug.
